internal/controller/ingress/tcpedge_controller.go

```diff
@@ -447,25 +449,53 @@ func (r *TCPEdgeReconciler) listTCPEdgesForIPPolicy(ctx context.Context, obj cli
 }
 
 func (r *TCPEdgeReconciler) updatePolicyModule(ctx context.Context, edge *ingressv1alpha1.TCPEdge, remoteEdge *ngrok.TCPEdge) error {
-	policy := edge.Spec.Policy
-	client := r.NgrokClientset.EdgeModules().TCP().RawPolicy()
+	if edge.Spec.Policy != nil && edge.Spec.TrafficPolicy != nil {
+		return fmt.Errorf("configuration contains both Policy and Traffic Policy")
+	}
+
+	client := r.NgrokClientset.EdgeModules().TCP().TrafficPolicy()
```

Comment:
Q. TCP here but policies only support HTTP. What's going on here?

Reply:
policies support TCP and TLS

Reply:
i know the gateway api only supports HTTP atm, but from what i understand, you can still use policy on TLS and TCP edges using the modules and/or the `NgrokTrafficPolicy` CRD

Reply:
there's also an `on_tcp_connect` phase :)

Reply:
we just don't have that phase explicitly written down anywhere within the operator. the only place that _needs_ the phases is the gateway API, since we convert the Filters into policy. Right now, Filters are only valid for HTTP edges, so those are the only phases we are concerned with atm

Reply:
for TCP/TLS edges using modules or the `NgrokTrafficPolicy`, we take their raw policy and just shove it into the api call. So for those edges, the user can use the `on_tcp_connect` phase

Reply:
fwiw it would be nice if we could completely delete awareness of phases from the operator, but I'm not sure if that's fully possible considering we need to translate the Filters. 

We will probably need to revisit this once we support TLS/TCP edges in the gateway

Reply:
acutally...
what do you think about representing the TrafficPolicy within the gateway api stuff as a `json.RawMessage` as well?

The upside is that we don't have to update the gateway api stuff every time there is a new phase. 
The downside is it's quite a bit messier, implementation wise. When resolving all the filters, we'd likely have to represent the policy document as `map[string]any` and dynamically merge the top-level keys.

Reply:
Ok, I actually went ahead and made this change. It decouples us from the deprecated Module definitions and (hopefully) lessens our future maintenance burden.